harness/tests/storage/test_storage_base.py

```diff
@@ -108,3 +109,41 @@ def test_shared_fs_shortcut_string() -> None:
 def test_bad_shortcut_string(shortcut: str) -> None:
     with pytest.raises(ValueError):
         _ = storage.from_string(shortcut)
+
+
+def test_azure_shortcut_dict() -> None:
```

Comment:
I really like how you took advantage of the refactored validation logic to give clear focus to your unit tests! :tada: